tests/tracer/test_hostname.py

```diff
@@ -1,14 +1,17 @@
 import mock
 
-from ddtrace.internal.hostname import get_hostname
+from ddtrace.internal import hostname
 
 
 @mock.patch("socket.gethostname")
 def test_get_hostname(socket_gethostname):
+    # Ensure _hostname is not set
+    hostname._hostname = None
```

Comment:
Since this relies on accessing a private variable, instead we should consider adding a `reset()` (or maybe `_reset()` is better?) helper method to the module.

Then 👍🏻